src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -320,42 +367,61 @@ void _stdcall TerminalUserScroll(void* terminal, int viewTop)
     publicTerminal->_terminal->UserScrollViewport(viewTop);
 }
 
-HRESULT _stdcall TerminalStartSelection(void* terminal, COORD cursorPosition, bool altPressed)
+HRESULT HwndTerminal::_StartSelection(LPARAM lParam) noexcept
 {
-    COORD terminalPosition = { cursorPosition };
+    try

```

Comment:
if you want, you can use function-level-try for these HRESULT functions and use the `CATCH_RETURN` macro to automatically turn any exception into an HRESULT.

```
HRESULT xyz() noexcept
try
{
}
CATCH_RETURN();
```

If you already did this and the SA carped at you, I accept that.

Reply:
this is true of every function with a single `RETURN_HR(wil::ResultFromCaughtException())` block.